pkg/proxy/nftables/proxier.go

```diff
@@ -1015,6 +1038,94 @@ func isAffinitySetName(set string) bool {
 	return strings.HasPrefix(set, servicePortEndpointAffinityNamePrefix)
 }
 
+// nftElementStorage is an internal representation of nftables map with single string value or set.
+type nftElementStorage struct {
+	elements      map[string]string
+	leftoverKeys  sets.Set[string]
+	containerType string
+	containerName string
+}
+
+func (c *nftElementStorage) joinKey(k []string) string {
+	return strings.Join(k, " . ")
+}
+
+func (c *nftElementStorage) splitKey(k string) []string {
+	return strings.Split(k, " . ")
+}
+
+// newNFTElementStorage creates an empty nftElementStorage and flushes the corresponding nftables container.
+func newNFTElementStorage(containerType, containerName string, tx *knftables.Transaction) *nftElementStorage {
+	c := &nftElementStorage{
+		elements:      make(map[string]string),
+		leftoverKeys:  sets.New[string](),
```

Comment:
this recreates these things in each iteration, golang has a Clear function now https://tip.golang.org/ref/spec#Clear so you don't need to reallocate memory